### PR TITLE
Gateway destinations

### PR DESCRIPTION
**Pull request by metalgearsloth**
Spawns planet gateways in a limited area with unlimited time to explore n stuff, we talked about it on discord like last month.

My other draft PRs got soaped 3 layers to hell so doing this in the interim and it's been on my list a while. Shouldn't need much work given biomes and salvage are the basis just need to make the gateway changes for playtesting.

Was around here we talked about it https://discord.com/channels/310555209753690112/770682801607278632/1140649274120613989

### CODE REVIEW

**Comment by metalgearsloth** [comment]
Video of the high-level overview:

https://github.com/space-wizards/space-station-14/assets/31366439/a9c41d82-f383-4d7b-be54-7b1315b8a4de

Still needs more polish but would rather get something in to playtest then kill expeditions in a subsequent PR and move the relevant code over.

File: Content.Server/Gateway/Systems/GatewayGeneratorSystem.cs [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    // Put salvage params in the UI
+
+    // Re-use salvage config stuff for the RNG
+    // Have it in the UI like expeditions.
+
+    // Also add weather coz it's funny.
+
+    // Add songs (incl. the downloaded one) to the ambient music playlist for planet probably.
+    // Copy most of salvage mission spawner
+
```
Comment (metalgearsloth) [inline Content.Server/Gateway/Systems/GatewayGeneratorSystem.cs:61]: This is a fat TODO but I mainly want to make sure people like the idea before killing expeds and moving it over here.

**Comment by Emisse** [comment]
conflicts

**Comment by metalgearsloth** [comment]
Conflicts sorted

**Pull request merged**